subsys/mpsl/init/mpsl_init.c

```diff
@@ -345,6 +349,80 @@ static void mpsl_calibration_work_handler(struct k_work *work)
 }
 #endif /* CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC && !CONFIG_SOC_SERIES_NRF54HX */
 
+#if !defined(CONFIG_SOC_SERIES_NRF54HX) && !defined(CONFIG_SOC_SERIES_NRF54LX)
+static uint16_t m_measure_hfxo_time_us(void)
+{
+	bool clock_running_on_boot = false;
+
+	if (nrf_clock_hf_is_running(NRF_CLOCK, NRF_CLOCK_HFCLK_HIGH_ACCURACY)
+		|| nrf_clock_start_task_check(NRF_CLOCK, NRF_CLOCK_DOMAIN_HFCLK)) {
+		clock_running_on_boot = true;
+	}
+
+	/* Make sure the HFCLK is off. */
+	nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_HFCLKSTOP);
+
+	nrf_timer_prescaler_set(NRF_TIMER0,
+		NRF_TIMER_PRESCALER_CALCULATE(NRF_TIMER_BASE_FREQUENCY_GET(NRF_TIMER0),
+			1000000));
+
+	nrf_timer_cc_set(NRF_TIMER0, 0, 400);
```

Comment:
I can update the comment above to say `/* Make sure the HFCLK is off by waiting for 400us. */`

Reply:
Isnt the crystal "hot" when you have stopped it, is 400 us sufficient for its oscillations to drop down? Typically, a connection interval is 1 second for HR service, the crystal will be "cold"-er compared to the 400 us used to measure the settling/started time?

1. Have we tested 4 second connection interval over the temperature range based on the measurement done?
2. Why such a design has not been there since the original design of clock start up?
3. Can we involve inputs from the h/w engineers to ensure we do not introduce regressions?